### PR TITLE
Add revert access for librarians

### LINKED ISSUE #2217: Give librarians revert access
### Is your feature request related to a problem? Please describe.
Librarians do not have the ability to revert changes such as botched redirects.

### PR DESCRIPTION
**Pull request by tapaswenipathak**
### Description

Fix #2217. [feature]

### Technical
Gives revert access for librarians.

### Testing
- Librarians should see history on top right side
- They can revert till a revision listed

### Evidence

- [ ] A little minute, adding a screenshot.

### CODE REVIEW

File: openlibrary/plugins/upstream/code.py [hunk 1]
```diff
@@ -191,7 +191,7 @@ def POST(self, key):
 
         user = accounts.get_current_user()
         is_admin = user and user.key in [m.key for m in web.ctx.site.get('/usergroup/admin').members]
-        if not (is_admin and web.ctx.site.can_write(key)):
+        if not (user.is_librarian() and is_admin and web.ctx.site.can_write(key)):
```
Comment (hornc) [inline openlibrary/plugins/upstream/code.py]: I think this should be `if not ((user.is_librarian() or is_admin) and web.ctx.site.can_write(key)):`
to allow either admins or librarians to revert.

Comment (tapaswenipathak) [reply openlibrary/plugins/upstream/code.py]: Added proper check for librarians and users.

Comment (tapaswenipathak) [reply openlibrary/plugins/upstream/code.py]: Added proper check for librarians and users as well, thanks for pointing that out.

**Comment by brad2014** [comment]
I've marked this pull request as P0, since it is associated with a P0 issue.  Is there anything blocking its push to production?

**Comment by cdrini** [comment]
Marking as "Needs: Feedback" until your screenshot is up :)

**Comment by cdrini** [comment]
@brad2014 I think just testing? As long as it's merged before Thursday (i.e. the deploy day) it should be fine.

**Comment by brad2014** [comment]
Thank you, @tapaswenipathak!

Just for housekeeping, I'm changing to `Priority 1` on the assumption that an out-of-cycle push is not required. Changing `Needs: Feedback` to `Needs: Review` on the assumption that the PR is presumed complete, and is waiting for @hornc or @mekarpeles  to approve and deploy (or or talk to @tapaswenipathak about what adjustments might be needed).

**Comment by cdrini** [comment]
@brad2014 Ah, that's my bad! I think `Needs: Detail` is more appropriate. @tapaswenipathak still has to add a screenshot (as noted in the description). The screenshot is proof that the PR author has tested the fix on their local machine. I'd rather wait for that before testing/reviewing it myself, since if it hasn't been tested, it will probably have bugs, and I only want to test/review this once :)

@tapaswenipathak Don't hesitate to ask here (or on slack) if you're having issues testing this locally!

**Comment by tapaswenipathak** [comment]
I am sorry folks, I completely missed the comments! I think facing issues in keeping track of everything, doing now.

**Comment by tapaswenipathak** [comment]
I am sorry folks, I completely missed the comments! doing now.

File: openlibrary/plugins/upstream/code.py [hunk 2]
```diff
@@ -191,7 +191,7 @@ def POST(self, key):
 
         user = accounts.get_current_user()
         is_admin = user and user.key in [m.key for m in web.ctx.site.get('/usergroup/admin').members]
```
Comment (hornc) [inline openlibrary/plugins/upstream/code.py]: I think this line can be removed now, and use the standard method `user.is_admin()` below on at line 194.

Other than this small (optional?) change, this PR looks good to me!

Comment (tapaswenipathak) [reply openlibrary/plugins/upstream/code.py]: Thanks, just did.

Comment (tapaswenipathak) [reply openlibrary/plugins/upstream/code.py]: Just did.

**Comment by mekarpeles** [comment]
@tapaswenipathak you rock!, thanks for driving this fix home

**Comment by tapaswenipathak** [comment]
Haha, lol.

**Comment by tapaswenipathak** [comment]
Haha, lol, I should for bunch of more open issues and PRs as well, brb, be there.

**Comment by tapaswenipathak** [comment]
Haha, lol, YES!! Happy you found it useful! BRB; on more of them.

**Comment by cdrini** [comment]
Hey @tapaswenipathak ! Sorry we had to revert your fix; it looked like it was causing an error #2261   :( But this is why testing is _so important_! You never know what could go wrong (especially when working with a new codebase!). We love helping new folks, so _please_ ask questions if you're having any trouble :)

**Comment by tapaswenipathak** [comment]
@cdrini - My bad, fixing that; high time.

**Pull request merged**